utils/updateReadme.ts

```diff
@@ -0,0 +1,87 @@
+import * as core from '@actions/core';
+import * as toolCache from '@actions/tool-cache';
+import * as setupGcloud from '@google-github-actions/setup-cloud-sdk';
+import path from 'path';
+import * as Moustache from 'mustache';
+import * as fs from 'fs';
+
+interface GcloudResponseObject {
+  output: string; // stringified json object
+  commandString: string; // the gcloud command string to be run
+}
+
+const setupGcloudSDK = async (): Promise<GcloudResponseObject> => {
+  try {
+    const cmd = ['functions', 'event-types', 'list', '--format', 'json'];
+    // Setup latest Gcloud Version
+    const gcloudVersion = await setupGcloud.getLatestGcloudSDKVersion();
+
+    if (!setupGcloud.isInstalled(gcloudVersion)) {
+      await setupGcloud.installGcloudSDK(gcloudVersion);
+    } else {
+      const toolPath = toolCache.find('gcloud', gcloudVersion);
+      core.addPath(path.join(toolPath, 'bin'));
+    }
+    await setupGcloud.authenticateGcloudSDK();
+    const authenticated = await setupGcloud.isAuthenticated();
+    if (!authenticated) {
+      throw new Error('Error authenticating the Cloud SDK.');
```

Comment:
Cant we use setup-gcloud as a pre step in the job that calls this so all of this is already setup?

Reply:
Will do!